app/src/main/java/com/arctouch/codingtest/RouteListActivity.java

```
@@ -0,0 +1,256 @@
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_route_list);
+
+        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
+        setSupportActionBar(toolbar);
+        toolbar.setTitle(getTitle());
+
+        RecyclerView recyclerView =(RecyclerView) findViewById(R.id.route_list);
+        assert recyclerView != null;
+
+        mAdapter = new QuerieRecyclerViewAdapter();
+        mAdapter.setListener(this);
+        recyclerView.setAdapter(mAdapter);
+
+
+        if (findViewById(R.id.pagerDetails) != null) {
+            mTwoPane = true;
+
+            mPagerAdapter = new DetailsFragmentPagerAdapter(getSupportFragmentManager(),null,this);
+            ViewPager pager = (ViewPager) findViewById(R.id.pagerDetails);
+            pager.setOffscreenPageLimit(4);
+            pager.setAdapter(mPagerAdapter);
+        }
+
+        handleIntent(getIntent());
+    }
+
+    private void handleIntent(Intent intent) {
+
+        if (Intent.ACTION_SEARCH.equals(intent.getAction())) {
+            String query = intent.getStringExtra(SearchManager.QUERY);
+            Log.i("Search Query", query);
+            SearchRecentSuggestions suggestions = new SearchRecentSuggestions(this,
+                    SuggestionsProvider.AUTHORITY, SuggestionsProvider.MODE);
+            suggestions.saveRecentQuery(query, null);
+
+            ArcTouchVolleyRequest jsRequest = findRoutesByNameRequest(query);
+            jsRequest.setTag(mAppRequestTag);
+            HttpHelper.getInstance(getApplicationContext()).addToRequestQueue(jsRequest);
+        }
+    }
+
+    private ArcTouchVolleyRequest findRoutesByNameRequest(String query) {
+        String servieUrl = "https://api.appglu.com/v1/queries/findRoutesByStopName/run";
+
+        JSONObject params = null;
+        try {
+            params = new JSONObject(String.format("{'params':{'stopName':'%s'}}","%"+query+"%"));
+        } catch (JSONException e) {
+            e.printStackTrace();
```

Yes there was, I could have used the Logger and also notified the user about the error. I intended to just discard the exception. A better approach also would not make a HTTP request with null parameters and maybe notify the user, Its a shame to admit but I overlooked this.